llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp

```diff
@@ -2121,6 +2120,31 @@ bool AArch64InstructionSelector::preISelLower(MachineInstr &I) {
     }
     return false;
   }
+  case TargetOpcode::G_INSERT_VECTOR_ELT: {
+    Register InsReg = I.getOperand(2).getReg();
```

Comment:
Please add a comment documenting what this is trying to do and why.

Reply:
I've added a comment and rebased - thanks.